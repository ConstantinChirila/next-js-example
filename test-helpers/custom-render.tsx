import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
