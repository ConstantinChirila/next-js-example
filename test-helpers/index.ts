export * from './custom-render'
export * from './setup'
