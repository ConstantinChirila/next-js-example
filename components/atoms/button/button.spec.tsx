import React from 'react'

import { Button } from './Button'
import { renderWithTheme } from '../../../test-helpers'

describe('Button', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(<Button>hello</Button>)
    expect(container).toMatchSnapshot()
  })
})
