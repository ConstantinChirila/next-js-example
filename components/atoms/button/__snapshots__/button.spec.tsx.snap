// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match a snapshot 1`] = `
.c0 {
  background-color: #452CDC;
  border-radius: 0.8rem;
  border: none;
  color: #FFFFFF;
  cursor: pointer;
  display: inline-block;
  font-family: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  height: 3.6rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  padding: 1.2rem;
  width: 100%;
}

<div>
  <button
    class="c0"
  >
    hello
  </button>
</div>
`;
